packages/experience-builder-sdk/src/hooks/useFetchExperience.ts

```diff
@@ -144,21 +159,32 @@ export const useFetchExperience = ({ mode, client }: useClientsideExperienceFetc
           setExperience(experience);
 
           return experience;
-        } catch (e) {
-          handleError(errorMessagesWhileFetching.experienceReferences, e);
-          throw e;
+        } catch (error) {
+          handleError(errorMessagesWhileFetching.experienceReferences, error);
+          setError(error as Error);
         }
-      } catch (e) {
-        handleError(errorMessagesWhileFetching.experience, e);
-        throw e;
+      } catch (error) {
+        handleError(errorMessagesWhileFetching.experience, error);
+        setError(error as Error);
       } finally {
         setIsFetching(false);
       }
     },
     [client, mode]
   );
 
+  useEffect(() => {
+    if (slug) {
+      fetchBySlug({ slug, localeCode, experienceTypeId });
+    } else if (id) {
+      fetchById({ id, localeCode, experienceTypeId });
+    } else {
+      setError(Error('Either slug or id must be provided to useFetchExperience'));
+    }
```

Comment:
Small code pattern opinion; I find early returns to be a more readable pattern compared to else ifs. It doesn't matter too much here since the benefits are mainly for nested ifs but I do like to try using it for all conditionals

Something like:
```ts
useEffect(() => {
  if (slug) {
    fetchBySlug({ slug, localeCode, experienceTypeId });
    return;
  }

  if(id) {
    fetchById({ id, localeCode, experienceTypeId });
    return;
  }

  setError(Error('Either slug or id must be provided to useFetchExperience'));
[]);
```